draft-ietf-jose-hpke-encrypt.md

```diff
@@ -345,11 +429,18 @@ HPKE also offers modes that offer authentication.
 HPKE relies on a source of randomness to be available on the device.
 In Key Agreement with Key Wrapping mode, CEK has to be randomly generated and it MUST be ensured that the guidelines in {{RFC8937}} for random number generations are followed.
 
+## HPKE authentication
+
+Authenticated HPKE modes MUST NOT be used with Key Encryption, as this is trivially insecure.
+
 ## Key Management
 
-Reusing a single KEM key across multiple algorithm combinations MUST be avoided to maintain cryptographic security. Each key and its associated algorithm suite, comprising the KEM, KDF, and AEAD, should be managed independently. This separation prevents unintended interactions or vulnerabilities between suites, ensuring the integrity and security guarantees of each algorithm suite are preserved.
-Additionally, the same key MUST NOT be used for both key wrapping and content encryption, as it may introduce security risks.  It
-creates algorithm confusion, increases the potential for key leakage, cross-suite attacks, and improper handling of the key.
+A single key SHOULD NOT be used as both authentication key and recipient key, as this has not been proven to be safe.
```

Comment:
Looking at HPKE RFC, I think it should be "sender key" instead of "authentication key".

Reply:
I can't find the term "sender key" in HPKE RFC.

Reply:
I got that from reading section 5.1.3...

The notation section implies that the terms are "sender role" and "recipient role". So presumably the statement should be:

A single key SHOULD NOT be used in both sender and recipient roles, as this has not been proven to be safe.

Reply:
Why would an application use the same key for sender and recipient roles ?
I don't see this attack discussed in HPKE.

Reply:
I guess application could use the same key for sender and recipient roles because it is easier. It is not good cryptographic practice.

Last paragraph of section 9.2.3. The paragraph talks about using key in multiple "modes", but the only possible issue is with using the same key in both sender and recipient roles. Most likely that is not an issue, but there is no proof of that.

Reply:
Agreed, I propose to update the text as follows::
A single key MUST NOT be used in both sender and recipient roles. Avoiding the use of the same key for both sender and recipient roles ensures clear cryptographic boundaries and minimizes unintended interactions.